src/xmipp/libraries/reconstruction/resolution_fso.cpp

```diff
@@ -25,1334 +25,1078 @@
+		MAT_ELEM(angles, 0, 69) = 108.000000;	 MAT_ELEM(angles, 1, 69) = 90.000000;
+		MAT_ELEM(angles, 0, 70) = 116.867209;	 MAT_ELEM(angles, 1, 70) = -75.219088;
+		MAT_ELEM(angles, 0, 71) = 99.132791;	 MAT_ELEM(angles, 1, 71) = -75.219088;
+		MAT_ELEM(angles, 0, 72) = 36.000000;	 MAT_ELEM(angles, 1, 72) = -26.565058;
+		MAT_ELEM(angles, 0, 73) = 48.811021;	 MAT_ELEM(angles, 1, 73) = -42.234673;
+		MAT_ELEM(angles, 0, 74) = 23.188979;	 MAT_ELEM(angles, 1, 74) = -42.234673;
+		MAT_ELEM(angles, 0, 75) = 54.466996;	 MAT_ELEM(angles, 1, 75) = -59.620797;
+		MAT_ELEM(angles, 0, 76) = 36.000000;	 MAT_ELEM(angles, 1, 76) = -58.282544;
+		MAT_ELEM(angles, 0, 77) = 17.533003;	 MAT_ELEM(angles, 1, 77) = -59.620797;
+		MAT_ELEM(angles, 0, 78) = 36.000000;	 MAT_ELEM(angles, 1, 78) = 90.000000;
+		MAT_ELEM(angles, 0, 79) = 44.867209;	 MAT_ELEM(angles, 1, 79) = -75.219088;
+		MAT_ELEM(angles, 0, 80) = 27.132791;	 MAT_ELEM(angles, 1, 80) = -75.219088;
+	}
 
-					if (cosine>=cosAngle)
-						{
-						if (f>maxFreq)
-						{
-							++n;
-							continue;
-						}
-						int idx = (int) round(f * m1sizeX);
-						cosine = exp( -((cosine -1)*(cosine -1))*aux );
-						DIRECT_MULTIDIM_ELEM(threeD_FSC, n) += cosine*dAi(fsc, idx);
-						DIRECT_MULTIDIM_ELEM(counterMap, n) += cosine;
-						}
-					++n;
-				}
-			}
-		}
-    }
+	angles *= PI/180;
+}
 
-    //TODO: Merge with Simple
-    void anistropyParameter(const MultidimArray<double> FSC,
-    		MultidimArray<double> &directionAnisotropy, size_t dirnumber,
-			MultidimArray<double> &aniParam, double thrs)
-    {
-    	double N = 0;
-		for (size_t k = 0; k<aniParam.nzyxdim; k++)
+
+void ProgFSO::anistropyParameter(const MultidimArray<double> FSC,
+		MultidimArray<double> &directionAnisotropy, size_t dirnumber,
+		MultidimArray<double> &	aniParam, double thrs)
+{
+	double N = 0;
+	for (size_t k = 0; k<aniParam.nzyxdim; k++)
+	{
+		if (DIRECT_MULTIDIM_ELEM(FSC, k) >= thrs)
 		{
-			if (DIRECT_MULTIDIM_ELEM(FSC, k) >= thrs)
-			{
-				DIRECT_MULTIDIM_ELEM(aniParam, k) += 1.0;
-				N++;
-			}
+			DIRECT_MULTIDIM_ELEM(aniParam, k) += 1.0;
+			N++;
 		}
-		DIRECT_MULTIDIM_ELEM(directionAnisotropy, dirnumber) = N;
-    }
+	}
+	DIRECT_MULTIDIM_ELEM(directionAnisotropy, dirnumber) = N;
+}
 
-    void anistropyParameterSimple(const MultidimArray<double> FSC,
-			MultidimArray<double> &aniParam, double thrs)
-    {
 
-		for (size_t k = 0; k<aniParam.nzyxdim; k++)
-			if (DIRECT_MULTIDIM_ELEM(FSC, k) >= thrs)
-				DIRECT_MULTIDIM_ELEM(aniParam, k) += 1.0;
-    }
+void ProgFSO::prepareData(MultidimArray<double> &half1, MultidimArray<double> &half2)
+{
+	Image<double> mask;
+	MultidimArray<double> &pmask = mask();
 
-    void prepareData(FileName &fnhalf1, FileName &fnhalf2,
-    		MultidimArray<double> &half1, MultidimArray<double> &half2, bool test)
-    {
-    	MultidimArray<double> &phalf1 = half1, &phalf2 = half2;
+	
+	std::cout << "Reading data..." << std::endl;
+	Image<double> imgHalf1, imgHalf2;
```

Comment:
use 
`Image(const MultidimArray<T> &im)`. You will avoid multidimarray copy online 960/961